core/src/main/java/hudson/PluginWrapper.java

```diff
@@ -534,6 +549,14 @@ public boolean hasLicensesXml() {
      *             thrown if one or several mandatory dependencies doesn't exists.
      */
     /*package*/ void resolvePluginDependencies() throws IOException {
+        String requiredCoreVersion = getRequiredCoreVersion();
+        if (requiredCoreVersion == null) {
+            throw new IOException(shortName + " doesn't declare required core version.");
```

Comment:
Hmm, not sure if this is even legal, but for the sake of compatibility perhaps this should be demoted to a warning?